governance/helpers/bridge/xCall.js

```diff
@@ -0,0 +1,153 @@
+const { ethers } = require('hardhat')
+const { ADDRESS_ZERO } = require('@unlock-protocol/hardhat-helpers')
+const { networks } = require('@unlock-protocol/networks')
+const IXCalled = require('./abis/IXCalled')
+
+const targetChains = Object.keys(networks)
+  .map((id) => networks[id])
+  .filter(
+    ({ governanceBridge, isTestNetwork, id }) =>
+      !isTestNetwork && !!governanceBridge && id != 1
+  )
+
+/***
+ * CONNEXT logic
+ */
+const getXCalledEvents = async (hash) => {
+  const { interface } = await ethers.getContractAt(IXCalled, ADDRESS_ZERO)
+  const { logs } = await ethers.provider.getTransactionReceipt(hash)
+  const parsedLogs = logs.map((log) => {
+    try {
+      return interface.parseLog(log)
+    } catch (error) {
+      return {}
+    }
+  })
+
+  const xCalled = parsedLogs
+    .filter((e) => e !== null)
+    .filter(({ name }) => name === 'XCalled')
+    .map(({ args }) => args)
+
+  return xCalled
+}
+
+const fetchOriginXCall = async ({ transferIds = [], chainId = 1 }) => {
+  const query = `
+    {
+      originTransfers(where:{
+        transferId_in: ${JSON.stringify(transferIds)}
+      }) {
+        chainId
+        nonce
+        transferId
+        to
+        delegate
+        receiveLocal
+        callData
+        slippage
+        originSender
+        originDomain
+        destinationDomain
+        transactionHash
+        bridgedAmt
+        status
+      }
+    }
+  `
+  const { originTransfers } = await fetchXCall({ query, chainId })
+  return originTransfers
+}
+
+const fetchDestinationXCall = async ({ transferIds, chainId }) => {
+  const query = `
+    {
+      destinationTransfers(where:{
+        transferId_in: ${JSON.stringify(transferIds)}
+      }) {
+        chainId
+        nonce
+        transferId
+        to
+        delegate
+        receiveLocal
+        callData
+        originDomain
+        destinationDomain
+        delegate
+        status
+        executedTransactionHash
+        reconciledTransactionHash
+      }
+    }
+  `
+  const { destinationTransfers } = await fetchXCall({ query, chainId })
+  return destinationTransfers
+}
+
+// supported chains by domain id
+const getSupportedChainsByDomainId = async () => {
+  return Object.keys(networks)
+    .map((id) => networks[id])
+    .filter(
+      ({ governanceBridge, isTestNetwork, id }) =>
+        !isTestNetwork && !!governanceBridge && id != 1
+    )
+    .reduce(
+      (prev, curr) => ({
+        ...prev,
+        [curr.governanceBridge.domainId]: curr,
+      }),
+      {}
+    )
+}
+
+const connextSubgraphIds = {
```

Comment:
Maybe we should mobe these to the networks file?

Reply:
not sure we will need them someone else than here for now, so better to keep them here imo